bigchaindb/lib.py

```diff
@@ -94,29 +95,23 @@ def write_transaction(self, transaction, mode):
 
     def _process_post_response(self, response, mode):
         logger.debug(response)
-        if response.get('error') is not None:
-            return (500, 'Internal error')
+
+        result = response['result']
+        if mode == self.mode_commit:
+            check_tx_code = result.get('check_tx', {}).get('code', 0)
```

Comment:
If you omit the second parameter, `get` returns `None` if the lookup is unsuccessful, so you can just write `check_tx_code = result.get('check_tx', {}).get('code')` if you want.

Reply:
`0` is the success code the API uses so I put it there but `None` would work here as well. I do not have a preference.